src/NuGet.Core/NuGet.Versioning/VersionRange.cs

```diff
@@ -365,5 +366,41 @@ public virtual string ToShortString()
         {
             return ToString("A", new VersionRangeFormatter());
         }
+
+        /// <summary>
+        /// Equals implementation for VersionRange.
+        /// </summary>
+        public bool Equals(VersionRange other)
+        {
+            if (other == null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return base.Equals(other) &&
+                IsFloating == other.IsFloating &&
+                EqualityUtility.EqualsWithNullCheck(Float, other.Float);
+        }
+
+        /// <summary>
+        /// Compare the obj as VersionRange.
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as VersionRange);
+        }
+
+        /// <summary>
+        /// Returns the VersionRangeBase hashcode
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
The hash code needs to consider the float range and do the equivalent to what Equals does.

Reply:
It doesn't have to... it just make them part of single bucket which is reasonable for our scenarios. 

Even in current codebase, we have a generic impl of `GetHashCode` in `VersionRangeBase` which is also reasonable.

Reply:
I'm not sure I follow. 
Both Equals/Hashcode should always work on the same members. 
And in this one that doesn't seem to be the case. 

Equals/Hashcode not being consistent can break hashset/dict scenarios.

Reply:
Discussed offline. I disagree with the arguments, so I'll let others/you take a call on this.

Reply:
@NuGet/nuget-client please let me know if anyone else has any concern with this, else I'll just merge it.

Reply:
I agree with @nkolev92 that is a good practice to have both Equals/Hashcode use the same members to compute its value.